Update and rename CHANGELOG to CHANGELOG.md

EwoutH opened the pull request:
Converts the Changelog to markdown syntax.

alecpl commented on CHANGELOG.md:
```diff
@@ -1016,11 +1017,11 @@ RELEASE 1.2-beta
 - Fix redundant blank lines when using HTML and top posting (#4927)
 - Fix redundant blank lines on start of text after html to text conversion (#4928)
-- Fix HTML sanitizer to skip <!-- node type X --> in output (#4932)
+- Fix HTML sanitizer to skip <! node type X > in output (#4932)
```
Changes like these aren't acceptable.

alecpl reviewed (changes requested):
Access to the new file will have to be blocked in the .htaccess file.

EwoutH commented:
@alecpl Thanks for your comments! I did should have used stricter requirements on my regex replace to remove the dashed lines (-----), that is now fixed. I also fixed a few `<!>` cases that got recognized as comments, and replaces a ' that should have been a `.

alecpl commented:
Please, resolve conflicts and I'll merge it.

EwoutH committed:
Update and rename CHANGELOG to CHANGELOG.md

Converts the Changelog to markdown syntax.
